### PR TITLE
Log index event requests on transport layer

### PR DESCRIPTION
**Pull request by sujithvm**
*Issue #, if available:*

*Description of changes:*

- Log index requests as separate category on Transport layer 
- These are subset of all requests on Transport layer with GRANTED_PRIVILEGES filtered by `indices:admin/` https://opendistro.github.io/for-elasticsearch-docs/docs/security/access-control/permissions/#indices 
- Enabling INDEX_EVENT and GRANTED_PRIVILEGES will result in 2 audit logs being generated for the same category
- Wrote tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java [hunk 1]
```diff
@@ -272,6 +272,21 @@ public void logGrantedPrivileges(String privilege, TransportRequest request, Tas
         }
     }
 
+    @Override
+    public void logIndexEvent(String privilege, TransportRequest request, Task task) {
+        if(!checkTransportFilter(AuditCategory.INDEX_EVENT, privilege, getUser(), request)) {
+            return;
+        }
+        // log only cluster admin action
+        if (!privilege.startsWith("indices:admin/")) {
```
Comment (debjanibnrj) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java:281]: Why are we ignoring non cluster admin?

File: src/test/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/DisabledCategoriesTest.java [hunk 2]
```diff
@@ -209,6 +209,8 @@ protected void logAll(AuditLog auditLog) {
 		logTransportBadHeaders(auditLog);
 		logTransportFailedLogin(auditLog);
 		logTransportSucceededLogin(auditLog);
+
```
Comment (debjanibnrj) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/DisabledCategoriesTest.java:212]: nit: empty line

File: src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java (hunk 1 above)
Comment (sujithvm) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java:281]: We are interested only in Index actions https://opendistro.github.io/for-elasticsearch-docs/docs/security/access-control/permissions/#indices

Comment (debjanibnrj) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java:281]: Index actions also include indices:data/* and indices:monitor/* options

Comment (sujithvm) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java:281]: indices:data involves document indexing and indices:monitor involves monitoring requests. 
We are only interested in requests affecting an index like index is created, deleted, settings updated, close, open, update mappings so on.

File: src/test/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/DisabledCategoriesTest.java (hunk 2 above)
Comment (sujithvm) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/DisabledCategoriesTest.java:212]: There are new line separations above it hence added.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java (hunk 1 above)
Comment (sujithvm) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java:281]: Also document data access is already covered under compliance

Comment (debjanibnrj) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java:281]: Thanks for the information.

**Pull request merged**